src/inverseRobot.js

```diff
@@ -24,7 +24,20 @@
  * @return {object}
  */
 function inverseRobot(robot) {
-  // write code here
+  let result = {};
+  const check = [...new Set(Object.values(robot))];
+
+  if (check.length === 0) {
+    result = {};
+  } else if (check.length !== Object.values(robot).length) {
```

Comment:
your solution is too complicated. to check that the object already has such a key, it is better to use hasOwnProperty:
[CODE STYLE] - use correct check for property presence in object. Some properties can be present, but still contain falsy value.
GOOD EXAMPLE:
if (key in robot) {

EVEN BETTER EXAMPLE:

if (result.hasOwnProperty(key))